client/components/data/document-head/index.jsx

```diff
@@ -65,6 +76,10 @@ class DocumentHead extends Component {
 				this.props.addMeta( meta );
 			} );
 		}
+
+		if ( nextProps.formattedTitle !== this.props.formattedTitle ) {
```

Comment:
I know `componentDidMount` doesn't fire in SSR, but does `componentWillReceiveProps` ? I'd imagine a component can't receive new props if it hasn't already been mounted, but worth checking to be safe.